src/main/java/teammates/common/util/Url.java

```diff
@@ -2,14 +2,12 @@
 
 public class Url {
 
+    protected final String baseUrl;
     protected String urlString;
 
-    public Url(String url) {
-        this.urlString = url.replace(getAppUrl(), ""); // force the URL to be relative
-    }
-
-    protected String getAppUrl() {
-        return Config.APP_URL;
+    public Url(String baseUrl, String url) {
```

Comment:
That will need an assumption that the URL is always in the form of `http://base.url/path`. `file:///C:/path/to/file` isn't, and it's even harder there to tell which one is the "base".
Also, the one-param constructor won't really fit well. If the one-param constructor is tailored for cases like this, then everything else should be called with two-param constructor. Then the base URL will be repeated for every call, i.e:
```java
new Url("http://www.google.com/page"); // this is cool
new Url(Config.APP_URL + "/page"); // but not this, Config.APP_URL is repeated
new Url(Config.APP_URL, "/page"); // and so is this
```